features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextSemanticsNodeMapper.kt

```diff
@@ -0,0 +1,148 @@
+                id = semanticsNode.id.toLong(),
+                x = bounds.x,
+                y = bounds.y,
+                width = bounds.width,
+                height = bounds.height,
+                text = text ?: "",
+                textStyle = textStyle,
+                textPosition = resolveTextAlign(semanticsNode)
+            ),
+            null
+        )
+    }
+
+    private fun resolveTextAlign(semanticsNode: SemanticsNode): MobileSegment.TextPosition? {
+        return resolveSemanticsTextStyle(semanticsNode)?.let {
+            val align = when (it.textAlign) {
+                TextAlign.Start,
+                TextAlign.Left -> MobileSegment.Horizontal.LEFT
+
+                TextAlign.End,
+                TextAlign.Right -> MobileSegment.Horizontal.RIGHT
+
+                TextAlign.Justify,
+                TextAlign.Center -> MobileSegment.Horizontal.CENTER
+
+                else -> MobileSegment.Horizontal.LEFT
+            }
+            MobileSegment.TextPosition(
+                alignment = MobileSegment.Alignment(
+                    horizontal = align
+                )
+            )
+        }
+    }
+
+    private fun resolveTextStyle(semanticsNode: SemanticsNode, parentContext: UiContext): MobileSegment.TextStyle? {
+        return resolveSemanticsTextStyle(semanticsNode)?.let { textStyle ->
+            val color = resolveModifierColor(semanticsNode) ?: textStyle.color
+            MobileSegment.TextStyle(
+                family = when (val value = textStyle.fontFamily) {
+                    is GenericFontFamily -> value.name
+                    else -> DEFAULT_FONT_FAMILY
+                },
+                size = textStyle.fontSize.value.toLong(),
+                color = convertColor(color.value.toLong()) ?: parentContext.parentContentColor
+                    ?: DEFAULT_TEXT_COLOR
+            )
+        }
+    }
+
+    private fun resolveModifierColor(semanticsNode: SemanticsNode): Color? {
+        val modifier = semanticsNode.layoutInfo.getModifierInfo().firstOrNull {
+            ComposeReflection.TextStringSimpleElement?.isInstance(it.modifier) ?: false
+        }?.modifier
+        modifier?.let {
+            if (ComposeReflection.TextStringSimpleElement?.isInstance(it) == true) {
+                val colorProducer = ComposeReflection.ColorProducerField?.getSafe(it) as? ColorProducer
+                return colorProducer?.invoke()
+            }
+        }
+        return null
+    }
+
+    private fun resolveSemanticsTextStyle(semanticsNode: SemanticsNode): TextStyle? {
+        val textLayoutResults = mutableListOf<TextLayoutResult>()
+        semanticsNode.config.getOrNull(SemanticsActions.GetTextLayoutResult)?.action?.invoke(textLayoutResults)
+        return textLayoutResults.firstOrNull()?.layoutInput?.style
+    }
+
+    private fun resolveText(semanticsConfiguration: SemanticsConfiguration): String? {
+        for ((key, value) in semanticsConfiguration) {
```

Comment:
minor: to minimize the number of `return` instructions it can be written as `return semanticsConfiguration.firstOrNull { it.key == KEY_CONFIG_TEXT }?.let { resolveAnnotatedString(it.value) }`

Reply:
updated